pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,617 @@
+			PodIPs: []v1.PodIP{
+				{
+					IP: "1.2.3.4",
+				},
+			},
+			Phase: v1.PodPending,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "pod1",
+			Namespace:         "pod1_ns",
+			UID:               "pod1",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+		},
+	}
+	pod2 = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "5.6.7.8",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod2",
+			Namespace: "pod2_ns",
+			UID:       "pod2",
+		},
+	}
+	pod3 = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "4.3.2.1",
+				},
+			},
+			Phase: v1.PodRunning,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "pod1",
+			Namespace:         "pod1_ns",
+			UID:               "pod1",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+		},
+	}
+	pod4 = &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "1.2.3.4",
+				},
+			},
+			Phase: v1.PodSucceeded,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "pod1",
+			Namespace:         "pod1_ns",
+			UID:               "pod4",
+			CreationTimestamp: metav1.Time{Time: refTime.Add(-5 * time.Minute)},
+			DeletionTimestamp: &metav1.Time{Time: refTime},
+		},
+	}
+	timestampMap = map[types.UID]*podLocalTimestamps{
+		"pod1": {CreationTimestamp: refTime},
+		"pod2": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+		"pod4": {CreationTimestamp: refTime.Add(-5 * time.Minute), DeletionTimestamp: &refTime},
+	}
+	node = &v1.Node{}
+)
+
+func Test_onPodUpdate(t *testing.T) {
+	newPod1 := &v1.Pod{
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{
+					IP: "4.5.6.7",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "pod1_ns",
+			UID:       "pod1",
+		},
+	}
+	tests := []struct {
+		name        string
+		podStore    *PodStore
+		oldObj      interface{}
+		newObj      interface{}
+		expectedPod *v1.Pod
+	}{
+		{
+			name: "newObj is not Pod",
+			podStore: &PodStore{
+				pods: cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			newObj:      node,
+			expectedPod: pod1,
+		},
+		{
+			name: "Valid case",
+			podStore: &PodStore{
+				pods: cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+			},
+			newObj:      newPod1,
+			expectedPod: newPod1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.podStore.pods.Add(pod1)
+			tt.podStore.onPodUpdate(tt.oldObj, tt.newObj)
+			assert.Equal(t, 1, len(tt.podStore.pods.List()))
+			assert.Equal(t, tt.expectedPod, tt.podStore.pods.List()[0].(*v1.Pod))
+		})
+	}
+}
+
+func Test_onPodCreate(t *testing.T) {
+	tests := []struct {
+		name                    string
+		podStore                *PodStore
+		obj                     interface{}
+		expectedTimestampMapLen int
+		expectedCreationTime    time.Time
```

Comment:
just replace this with `expectedMap` and have a single check in the test?

Reply:
Fixed. Thanks.